### PR TITLE
Docs for Profiling

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**

### CODE REVIEW

File: website/docs/tutorials/09-profiling.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+sidebar_label: Profiling
+---
+
+# Profiling
+
+## Prerequisites
+
+You need pprof installed beforehand. You can learn how to install it [HERE](https://github.com/google/pprof#building-pprof)
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:9]:
```suggestion
You need pprof installed beforehand.
You can learn how to install it [in pprof documentation](https://github.com/google/pprof#building-pprof).
```

File: website/docs/tutorials/09-profiling.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+In case you have a test with high number of steps/memory holes, Protostar can help you diagnose which functions are the most expansive in terms of those resources.
+
+If you want to generate profile for a test case, run:
+
+```shell
+protostar test --profiling test/test_file.cairo::test_case_name 
+```
+:::warning
+You can only run profiling for a single test case
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:21]:
```suggestion
You can only run profiling for a single test case.
```

File: website/docs/tutorials/09-profiling.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+:::
+
+Protostar will run the test in the profiling mode (it may take a little more than ususal) and produce a file `profile.pb.gz`
+
+Then you can read the profile using:
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:26]: I wonder if it would be possible to bundle pprof with Protostar 🤔

Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/09-profiling.md:26]: I wonder as well, we can investigate it later

Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/09-profiling.md:26]: https://github.com/software-mansion/protostar/issues/986

**Review by mkaput — CHANGES_REQUESTED** [review]
I miss a showcase of what the profiler is capable of doing, like telling people that we profile memholes etc.

**Comment by MaksymilianDemitraszek** [comment]
@mkaput 
> In case you have a test with high number of steps/memory holes, Protostar can help you diagnose which functions are the most expansive in terms of those resources.

I will make it more explicit

**Comment by mkaput** [comment]
hmm I was more thinking about adding an explicit list of profiled properties

**Review by mkaput — CHANGES_REQUESTED** [review]
`profile.pb.gz` seems like something which should not land in the repo

File: website/docs/tutorials/09-profiling.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+go tool pprof -http=":8000" profile.pb.gz
+```
+### Example results
+
+![Profiler](/img/prof1.jpg)
+![Profiler](/img/prof2.jpg)
+![Profiler](/img/prof3.jpg)
+
+
+
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:41]:
```suggestion
```

File: website/docs/tutorials/09-profiling.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Protostar will run the test in the profiling mode (it may take a little more than ususal) and produce a file `profile.pb.gz`
+
+Then you can read the profile using: 
+```shell
+go tool pprof -http=":8000" profile.pb.gz
+```
+### Example results
+
+![Profiler](/img/prof1.jpg)
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:36]: you can colocate images with markdown: https://docusaurus.io/docs/markdown-features/assets

File: website/docs/tutorials/09-profiling.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+---
+sidebar_label: Profiling
+---
+
```
Comment (mkaput) [inline website/docs/tutorials/09-profiling.md:4]:
```suggestion
```

header already does this

File: website/docs/tutorials/09-profiling.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+# Profiling
+
+## Prerequisites
+
+You need pprof installed beforehand.
```
Comment (Arcticae) [inline website/docs/tutorials/09-profiling.md:9]: I think we should mention that this is only needed for interpreting the results

**Review by Arcticae — APPROVED** [review]

**Comment by Arcticae** [comment]
Please address Marek's comments though

File: website/docs/tutorials/09-profiling.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+Protostar will run the test in the profiling mode (it may take a little more than ususal) and produce a file `profile.pb.gz`
+
+Then you can read the profile using: 
+```shell
+go tool pprof -http=":8000" profile.pb.gz
+```
+### Example results
+
+![Profiler](/img/prof1.jpg)
+![Profiler](/img/prof2.jpg)
```
Comment (kasperski95) [inline website/docs/tutorials/09-profiling.md]: [Style] This image could be scaled down to match the scaling used in `img/prof1.jpg`.

Comment (kasperski95) [reply website/docs/tutorials/09-profiling.md]: [Style] Images could be trimmed, and this document converted to mdx to control image scales from the code.

File: website/docs/tutorials/09-profiling.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+You can learn how to install it [in pprof documentation](https://github.com/google/pprof#building-pprof).
+
+## How to profile a contract?
+
+Protostar can help you diagnose which functions are the most expensive in terms of used resources.
+Protostar currently measures:
+- number of steps
+- memory holes
+
+If you want to generate profile for a test case, run:
```
Comment (kasperski95) [inline website/docs/tutorials/09-profiling.md:15]:
```suggestion
If you want to generate a profile for a test case, run:
```
What is a profile?

File: website/docs/tutorials/09-profiling.md [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+```shell
+protostar test --profiling test/test_file.cairo::test_case_name 
+```
+:::warning
+You can only run profiling for a single test case.
+:::
+
+Protostar will run the test in the profiling mode (it may take a little more than ususal) and produce a file `profile.pb.gz`
```
Comment (kasperski95) [inline website/docs/tutorials/09-profiling.md:24]:
```suggestion
Protostar will run the test in the profiling mode (it may take a little more than usual) and produce a file `profile.pb.gz`
```

File: website/docs/tutorials/09-profiling.md (hunk 9 above)
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/09-profiling.md:15]: I think it is a common term, I don't think we should explain it. The same way we do not explain what terminal is.

Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/09-profiling.md:15]: I think it is obvious from context what profile is. We have a sentence 
> Protostar can help you diagnose which functions are the most expensive in terms of used resources.

> A statistical summary of the events observed.

I do not think this is the meaning of word statistical. 

We could rephrase it to 
>Protostar can perform dynamic program analysis to help you diagnose which functions are the most expensive in terms of used resources.
 
but I am not sure if it is not unnecessary wordy. @mkaput WDYT?

**Review by mkaput — APPROVED** [review]

File: website/docs/tutorials/09-profiling.md (hunk 8 above)
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/09-profiling.md]: I do not want to dig into using html in docs in this pr, created https://github.com/software-mansion/protostar/issues/1012
